src/pages/proposals/new-index.tsx

```diff
@@ -0,0 +1,128 @@
+import { observer } from 'mobx-react-lite';
+import React, { FunctionComponent, useState } from 'react';
+import variables from 'src/utils/variables';
+import { RootState } from 'src/reducers/store';
+import { Loader } from 'src/components/common/loader';
+import { useStore } from 'src/stores';
+import { useAppSelector } from 'src/hooks/use-app-select';
+import NewCards from './new-cards';
+
+const selector = (state: RootState) => {
+	return {
+		lang: state.language,
+	};
+};
+
+enum ProposalStatuses {
+	all = 1,
+	pending,
+	passed,
+	rejected,
+}
+
+const ProposalsPage: FunctionComponent = observer(() => {
+	const { chainStore, queriesStore } = useStore();
+	const queries = queriesStore.get(chainStore.current.chainId);
+	const proposals = queries.cosmos.queryGovernance.proposals;
+
+	const { lang } = useAppSelector(selector);
+	const [active, setActive] = useState(ProposalStatuses.all);
+	const [filter, setFilter] = useState<number | null>(ProposalStatuses.all);
+
+	const handleChange = (value: number) => {
+		if (active === value) {
+			return;
+		}
+
+		setActive(value);
+		setFilter(value);
+	};
+	const filteredProposals =
+		filter !== ProposalStatuses.all ? proposals.filter(item => item.proposalStatus === filter) : proposals;
+	const getProposalStatusCount = (status: number) => {
```

Comment:
Instead of calculating this each render, you could use `useMemo` and calculate each count in one loop

```
const counts = useMemo(() => {
   let passedCount = 0;

  proposals.forEach((proposal) => {
      if (proposal.proposalStatus === ProposlalStatuses.passed) {
         passedCount++;
     }
  });

  return {
     passedCount,
   };
}, [proposals]);
```